components/brave_shields/browser/csp_merge_unittest.cc

```diff
@@ -0,0 +1,72 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_shields/browser/ad_block_service_helper.h"
+#include "testing/gmock/include/gmock/gmock.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace brave_shields {
+
+using ::testing::_;
+
+class CspMergeTest : public testing::Test {
+ public:
+  CspMergeTest() {}
+  ~CspMergeTest() override {}
+
+ protected:
+  void SetUp() override {}
+
+  void TearDown() override {}
+};
+
+const base::Optional<std::string> NO_POLICY = base::nullopt;
+
+const auto POLICY1 =
+    base::Optional<std::string>("script-src 'self' 'unsafe-inline'");
+const auto POLICY2 =
+    base::Optional<std::string>("media-src 'self' https://example.com");
+
+TEST_F(CspMergeTest, MergeTwoEmptyPolicies) {
+  const auto a = NO_POLICY;
+  auto b = NO_POLICY;
+
+  MergeCspDirectiveInto(a, &b);
+
+  ASSERT_EQ(b, NO_POLICY);
+}
+
+TEST_F(CspMergeTest, MergeEmptyIntoNonEmpty) {
+  const auto a = POLICY1;
+  auto b = NO_POLICY;
+
+  MergeCspDirectiveInto(a, &b);
+
+  ASSERT_EQ(b, POLICY1);
```

Comment:
personally I'd put all checks into a single test, just to minimize boilerplate

Reply:
It's not so simple because `MergeCspDirectiveInto` mutates one of the arguments. We'd still need the setup for `a` and `b` in each case; at that point the boilerplate amounts to the `TEST` header for each one and I think that's a reasonable tradeoff for independently runnable test cases.